commit 930106855f3f52dee01ba44b5d41a33974dfd26e
Author: Alexandre Julliard <julliard@winehq.org>
Date:   Thu Feb 27 11:04:53 2025 +0100

    makefiles: Link to compiler-rt for builds in MSVC mode.

diff --git a/tools/makedep.c b/tools/makedep.c
index a423bfd0d6f..3fe2d254fdf 100644
--- a/tools/makedep.c
+++ b/tools/makedep.c
@@ -159,6 +159,7 @@ static const char *ln_s;
 static const char *sed_cmd;
 static const char *wayland_scanner;
 static const char *sarif_converter;
+static const char *compiler_rt;
 static int so_dll_supported;
 static int unix_lib_supported;
 /* per-architecture global variables */
@@ -2296,6 +2297,7 @@ static struct strarray get_default_imports( const struct makefile *make, struct
         for (i = 0; i < imports.count; i++)
             if (!strcmp( imports.str[i], "winecrt0" )) return ret;
         strarray_add( &ret, "winecrt0" );
+        if (compiler_rt) strarray_add( &ret, compiler_rt );
         return ret;
     }
 
@@ -2304,6 +2306,7 @@ static struct strarray get_default_imports( const struct makefile *make, struct
             crt_dll = imports.str[i];
 
     strarray_add( &ret, "winecrt0" );
+    if (compiler_rt) strarray_add( &ret, compiler_rt );
     if (crt_dll) strarray_add( &ret, crt_dll );
 
     if (make->is_win16 && (!make->importlib || strcmp( make->importlib, "kernel" )))
@@ -4677,6 +4680,7 @@ int main( int argc, char *argv[] )
     ln_s               = get_expanded_make_variable( top_makefile, "LN_S" );
     wayland_scanner    = get_expanded_make_variable( top_makefile, "WAYLAND_SCANNER" );
     sarif_converter    = get_expanded_make_variable( top_makefile, "SARIF_CONVERTER" );
+    compiler_rt        = get_expanded_make_variable( top_makefile, "COMPILER_RT_PE_LIBS" );
 
     if (root_src_dir && !strcmp( root_src_dir, "." )) root_src_dir = NULL;
     if (tools_dir && !strcmp( tools_dir, "." )) tools_dir = NULL;
diff --git a/tools/winegcc/winegcc.c b/tools/winegcc/winegcc.c
index 4835e91aaed..600bd171b89 100644
--- a/tools/winegcc/winegcc.c
+++ b/tools/winegcc/winegcc.c
@@ -1200,6 +1200,8 @@ static void build(struct options* opts)
         add_library(opts, lib_dirs, &files, "advapi32");
         add_library(opts, lib_dirs, &files, "user32");
         add_library(opts, lib_dirs, &files, "winecrt0");
+        if (opts->target.platform == PLATFORM_WINDOWS)
+            add_library(opts, lib_dirs, &files, "compiler-rt");
         if (opts->use_msvcrt)
         {
             if (!crt_lib)
